inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -436,3 +439,117 @@ def update_system(self) -> None:
+            format(self.sota_cmd))
+        try:
+            content = get_os_version()
+            if content == UNKNOWN:
+                raise SotaError("Failed to get os version.")
+            state: dispatcher_state.DispatcherState
+            if dispatcher_state.is_dispatcher_state_file_exists():
+                consumed_state = dispatcher_state.consume_dispatcher_state_file(read=True)
+                restart_reason = None
+                if consumed_state:
+                    restart_reason = consumed_state.get('restart_reason', None)
+                if restart_reason:
+                    state = {'tiberos-version': content}
+            else:
+                state = (
+                    {'restart_reason': "sota",
+                     'tiberos-version': content}
+                )
+            dispatcher_state.write_dispatcher_state_to_state_file(state)
+        except (DispatcherException, SotaError) as err:
+            self._dispatcher_broker.telemetry(
+                "...state file creation unsuccessful.")
+            raise SotaError('Failed to create a dispatcher state file. Error: ', err)
+
+        self._dispatcher_broker.telemetry(
+            "Dispatcher state file creation successful.")
+
+    def commit(self) -> int:
+        """On TiberOS, this method runs a UT commit command
+
+        Also, delete dispatcher state file.
+        """
+        logger.debug("")
+        dispatcher_state.clear_dispatcher_state()
+        code = update_tool_commit_command()
+
+        return code
+
+    def recover(self, rebooter: Rebooter, time_to_wait_before_reboot: int) -> None:
+        """Recover from a failed SOTA.
+
+        On TiberOS, no action is required other than deleting the
+        state file and rebooting.
+        @param rebooter: Object implementing reboot() method
+        @param time_to_wait_before_reboot: If we are rebooting, wait this many seconds first.
+        """
+        logger.debug("")
+        dispatcher_state.clear_dispatcher_state()
+        time.sleep(time_to_wait_before_reboot)
+        rebooter.reboot()
+
+    def revert(self, rebooter: Rebooter, time_to_wait_before_reboot: int) -> None:
+        """Revert after second system SOTA boot when we see a problem with startup.
+
+        On TiberOS, we just reboot without commit
+        @param rebooter: Object implementing reboot() method
+        @param time_to_wait_before_reboot: If we are rebooting, wait this many seconds first.
+        """
+        logger.debug("time_to_wait_before_reboot = " + str(time_to_wait_before_reboot))
+        dispatcher_state.clear_dispatcher_state()
+        time.sleep(time_to_wait_before_reboot)
+        rebooter.reboot()
+
+    def update_system(self) -> None:
+        """If the system supports it, check whether the system was updated, after rebooting.
+
+        For TiberOS, we compare the image's SHA stored in dispatcher state file and current SHA.
```

Comment:
we should distinguish the SHA from the actual TiberOS version by called it `tiberos-image-sha` rathert han `tiberos-version`

Reply:
Instead of checking the image sha, Anuj suggests that we could check the version stored in /etc/os-release, which has the value like VERSION="2.0.20240802.0213". 
Will update the docstring.